src/Teams/Data/IRepository.cs

```diff
@@ -3,12 +3,12 @@
 
 namespace Teams.Data
 {
-    interface IRepository<T> where T : class, IQueryable
+    interface IRepository<T> where T : class
     {
-        T GetAll();
-        T GetById(T id);
-        Task<T> Insert(T entity);
-        Task<T> Update(T entity);
-        Task<T> Delete(T entity);
+        IQueryable<T> GetAll();
+        T GetById(T Id);
```

Comment:
Are we expecting that the id will be the same type as the entity?

Reply:
No. Changed return type to IQueryable<T>